inc/utils/namespace.php

```diff
@@ -235,6 +235,31 @@ function milliseconds() : int {
 	return (int) microtime( true ) * 1000;
 }
 
+/**
+ * Get past point in time in milliseconds, optionally rounded to the nearest time block.
+ *
+ * @param string $point_in_time strtotime-safe string, eg: '-1 week'
+ * @param integer $round_to Round the result to the nearest time block in seconds, eg: HOUR_IN_SECONDS.
+ *
+ * @return integer|null
+ */
+function time_ago_in_milliseconds( string $point_in_time, int $round_to = 0 ) : ?int {
+	$time_ago = strtotime( $point_in_time );
+
+	if ( ! $time_ago ) {
+		trigger_error( 'Analytics: Point in time string cannot be resolved.', E_USER_WARNING );
```

Comment:
We could make the error message a bit friendlier by adding in the value that was passed, would likely help with debugging.